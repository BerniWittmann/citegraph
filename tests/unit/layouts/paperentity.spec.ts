import { shallowMount } from '@vue/test-utils'
import { i18n } from '../setupPlugins'

import PaperEntityLayout from '@/layouts/PaperEntity.vue'

describe('layouts/PaperEntity.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(PaperEntityLayout, {
      i18n,
      slots: {
        title: '<h1 class="title">Title</h1>',
        default: '<p class="content">Main Content</p>'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders an edit button', () => {
    const wrapper = shallowMount(PaperEntityLayout, {
      i18n,
      slots: {
        title: '<h1 class="title">Title</h1>',
        default: '<p class="content">Main Content</p>'
      }
    })
    const btn = wrapper.find('v-btn-stub')
    expect(btn.exists()).toBeTruthy()
    expect(btn.text()).toContain('project.explore.view.edit')
  })
})
