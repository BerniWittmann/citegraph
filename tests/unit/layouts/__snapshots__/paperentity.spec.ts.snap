// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts/PaperEntity.vue renders 1`] = `
<v-container-stub tag="div">
  <v-card-stub loaderheight="4" tag="div" class="pa-4 paper-entity-card">
    <v-card-title-stub class="headline">
      <h1 class="title">Title</h1>
      <v-spacer-stub></v-spacer-stub>
      <v-btn-stub tag="button" activeclass="" to="[object Object]" type="button">
        <v-icon-stub>mdi-pencil</v-icon-stub>
        project.explore.view.edit
      </v-btn-stub>
    </v-card-title-stub>
    <p class="content">Main Content</p>
  </v-card-stub>
</v-container-stub>
`;
